src/clr-angular/forms/datepicker/providers/date-io.service.ts

```diff
@@ -41,6 +43,19 @@ export class DateIOService {
             // everything else is set to BIG-ENDIAN FORMAT
             this.localeDisplayFormat = BIG_ENDIAN;
         }
+        this.extractDelimiters();
+    }
+
+    private extractDelimiters(): void {
+        if (this.cldrLocaleDateFormat) {
+            // Sanitize Date Format. Remove RTL characters.
+            // FIXME: When we support RTL, remove this and handle it correctly.
+            const localeFormat: string = this.cldrLocaleDateFormat.replace(RTL_REGEX, "");
+            const delimiters: string[] = localeFormat.split(DELIMITER_REGEX);
+            if (delimiters.length === 4) {
+                this.delimiters = [delimiters[1], delimiters[2]];
+            }
```

Comment:
Just in case, do you mind logging or raising an error here? If we read more than 4 splits, then we have an issue somewhere and we'll want to fix it.

Reply:
I don't get what's going on here :P . Why is it including delimiter strings in an array? What value does it have over using just plain string delimiter? 

Why for example this:
`this.delimiters = [delimiters[1], delimiters[2]];` is not `this.delimeters = delimiters`?

Reply:
You are right. I should have added a comment here. Let me update the PR with a comment.

Explanation:
https://github.com/angular/angular/blob/master/packages/common/locales/kkj.ts#L46

![image](https://user-images.githubusercontent.com/1426805/37937347-45473e70-3127-11e8-9e5f-875f7a3595b5.png)

The date format received from CLDR is always in D..M..Y or M..D..Y or Y..M..D format. Our regex detects that and splits around the D, M and Y. The result is always an array with 4 strings.

Reply:
Also, @youdz pointed out on our internal chat that there is missing null value check here. Will fix that too.